src/components/is-scrolling-up.mixin.ts

```diff
@@ -0,0 +1,38 @@
+import { ScrollDirection, XOn } from '@empathyco/x-components';
+import Vue from 'vue';
+import Component from 'vue-class-component';
+@Component
+export default class IsScrollingUp extends Vue {
+  protected stopsScrollDown = true;
+  protected isScrollingUp = true;
+  protected scrollOffset = 200;
+
+  @XOn('UserChangedScrollDirection')
+  scrollDirectionChanged(direction: ScrollDirection): void {
+    this.stopsScrollDown = direction === 'UP';
+  }
+
+  @XOn(['UserScrolled'])
+  scrollLength(position: number): void {
+    // TODO change this implementation when the scroll module is fixed
+    if (this.$x.scroll['main-scroll']?.hasAlmostReachedEnd) {
+      this.isScrollingUp = false;
+      return;
+    }
+    if (this.stopsScrollDown || position < this.scrollOffset) {
+      this.isScrollingUp = true;
+    } else if (!this.stopsScrollDown && position > this.scrollOffset) {
+      this.isScrollingUp = false;
+    }
```

Comment:
Use the main scroll state and apply the changes to the property of `isScrollingUp`.

```suggestion
   const isScrollingUp = this.scrollPositionsMap['main-scroll']?.direction === 'UP';

    if (isScrollingUp || this.mainScrollPosition < this.scrollOffset) {
      this. hasScrolledPastThreshold = false;
    } else if (!isScrollingUp && this.mainScrollPosition > this.scrollOffset) {
      this. hasScrolledPastThreshold = true;
    }```